plugins/modules/vm_clone.py

```diff
@@ -82,19 +86,46 @@
 from ..module_utils.client import Client
 from ..module_utils.rest_client import RestClient
 from ..module_utils.vm import VM
+from ..module_utils.vm_snapshot import VMSnapshot as Snapshot
 from ..module_utils.task_tag import TaskTag
 
 
+def get_snapshot(
+    module: AnsibleModule, rest_client: RestClient, virtual_machine_obj: VM
+) -> AnsibleModule:
+    # Get snapshot from label.
+    if module.params["snapshot_label"]:
+        snapshot_list = Snapshot.get_snapshots_by_query(
+            dict(
+                label=module.params["snapshot_label"],
+                domainUUID=virtual_machine_obj.uuid,
+            ),
+            rest_client,
+        )
+        if not snapshot_list or not snapshot_list[0].get("snapshot_uuid"):
```

Comment:
You cannot assume snapshot_label is unique. Check snapshot_list has single element.